apps/root/src/common/components/contact-modal/edit-contact-modal/index.tsx

```diff
@@ -77,15 +81,22 @@ const EditContactModal = ({ contact, setActiveModal }: AddContactModalProps) =>
           fullWidth
         />
         <TextField id="editContactAddress" disabled value={contact.address} fullWidth type="text" />
-      </ContainerBox>
-      <Button variant="contained" size="large" onClick={onEditContact} fullWidth disabled={isLoading}>
-        {isLoading ? (
-          <CenteredLoadingIndicator size={32} />
-        ) : (
-          <FormattedMessage description="update" defaultMessage="Update" />
+      </StyledInputsContainer>
+      <ContainerBox gap={6}>
+        {goBack && (
+          <Button variant="outlined" size="large" onClick={goBack} fullWidth>
+            <FormattedMessage description="add-contact-modal.buttons.cancel" defaultMessage="Cancel" />
+          </Button>
         )}
-      </Button>
-    </ContainerBox>
+        <Button variant="contained" size="large" onClick={onEditContact} fullWidth disabled={isLoading}>
+          {isLoading ? (
+            <CenteredLoadingIndicator size={32} />
```

Comment:
We have an inconsistency in our user experience here. As we discussed some time ago, Contact Modal should not have a loading state, it should be closed and then the notification will confirm de operation status (I think this is done for creation, but awaited for loading)

Reply:
I get that, I don't think its something that we should tackle now, we can take it up with the ux designer